### ISSUE TITLE
Add the option, to reparent a node, at a specific child index

### ISSUE THREAD

**Issue by JHDev2006** [godotengine/godot-proposals#11364]
### Describe the project you are working on

Deckbuilder, which has balatro mechanics allowing you to move around your cards to organise them, and can transfer them between containers

### Describe the problem or limitation you are having in your project

When transferring a card to another container, it first reparents it, and then moves it to the specific slot, however, because i have some animations allowing the cards to rotate and move freely between positions, it creates a weird jitter effect, as the all the cards get updated twice. Once when the card gets reparented, and is added to the bottom of the child tree, and again, when it gets moved to the correct position.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

being able to allow for us to reparent to a specific child index as well, would mean i wouldnt have this weird jitter effect, and would create a really smooth feel!

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Add another argument onto the reparent() function, called idx, which would allow you to specify which child position the node will get added to, and will update the rest of the parented children, essentially, utilizing the move_child function aswell, the default value would be -1, which would simply add it to the bottom of the tree, as it does now, however setting the value to 0, would reparent it at the top of the child tree!

### If this enhancement will not be used often, can it be worked around with a few lines of script?

not currently, using call deferred, simply causes the reparent  to not work correctly, and awaiting a frame, causes a worse jitter effect

### Is there a reason why this should be core and not an add-on in the asset library?

it would be a pretty simple addition, but it would be used to have finer control over the order of control nodes, in a way that seems in line with the design philosophy of godot.

**Comment by Mickeon** [comment]
How would this be any different than calling `Node.reparent()` followed by calling `move_child()` on the parent node? That would be what happens internally if the proposal is implemented, which would still not solve your issue, I believe.

**Comment by JHDev2006** [comment]
Problem is, when it comes to using control nodes and containers, if I add the child control to a container, it then gets added to the bottom of the container, which is when I then have to move it, which happens after, but since I have some sprites, which lerp to the control nodes position, it creates a weird jitter effect, as all other nodes move to where it would be at the bottom, and then to where I WANT it to be

**Comment by Archeidos** [comment]
> Problem is, when it comes to using control nodes and containers, if I add the child control to a container, it then gets added to the bottom of the container, which is when I then have to move it, which happens after, but since I have some sprites, which lerp to the control nodes position, it creates a weird jitter effect, as all other nodes move to where it would be at the bottom, and then to where I WANT it to be

Yep, I second this - I use a custom interpolated container (a [plugin](https://github.com/don-tnowe/godot-extra-controls) by [don-tnowe](https://github.com/don-tnowe)) and if I don't do `move_child` after my reparenting, then my `Control` will lerp to the bottom of the container...

However, because `reparent` doesn't support positioning, the end result of having to do `move_child` causes jittering afterwards.
